flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/sink/writer/FileWriterBucket.java

```diff
@@ -0,0 +1,329 @@
+			if (LOG.isDebugEnabled()) {
+				LOG.debug(
+						"closing in-progress part file for bucket id={} due to element {}.",
+						bucketId,
+						element);
+			}
+
+			inProgressPart = rollPartFile(now);
+		}
+
+		inProgressPart.write(element, now);
+	}
+
+	List<FileSinkCommittable> prepareCommit(boolean flush) throws IOException {
+		if (inProgressPart != null && (rollingPolicy.shouldRollOnCheckpoint(inProgressPart)
+				|| flush)) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug(
+						"Closing in-progress part file for bucket id={} on checkpoint.",
+						bucketId);
+			}
+			closePartFile();
+		}
+
+		List<FileSinkCommittable> committables = new ArrayList<>();
+		pendingFiles.forEach(pendingFile -> committables.add(new FileSinkCommittable(pendingFile)));
+		pendingFiles.clear();
+
+		if (inProgressFileToCleanup != null) {
+			committables.add(new FileSinkCommittable(inProgressFileToCleanup));
+			inProgressFileToCleanup = null;
+		}
+
+		return committables;
+	}
+
+	FileWriterBucketState<BucketID> snapshotState() throws IOException {
+		InProgressFileWriter.InProgressFileRecoverable inProgressFileRecoverable = null;
+		long inProgressFileCreationTime = Long.MAX_VALUE;
+
+		if (inProgressPart != null) {
+			inProgressFileRecoverable = inProgressPart.persist();
+			inProgressFileToCleanup = inProgressFileRecoverable;
+			inProgressFileCreationTime = inProgressPart.getCreationTime();
+		}
+
+		return new FileWriterBucketState<>(
+				bucketId,
+				bucketPath,
+				inProgressFileCreationTime,
+				inProgressFileRecoverable);
+	}
+
+	private InProgressFileWriter<IN, BucketID> rollPartFile(long currentTime) throws IOException {
+		closePartFile();
+
+		final Path partFilePath = assembleNewPartPath();
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Opening new part file \"{}\" for bucket id={}.",
+					partFilePath.getName(), bucketId);
+		}
+
+		return bucketWriter.openNewInProgressFile(bucketId, partFilePath, currentTime);
+	}
+
+	/**
+	 * Constructor a new PartPath and increment the partCounter.
+	 */
+	private Path assembleNewPartPath() {
+		long currentPartCounter = partCounter++;
+		return new Path(
+				bucketPath,
+				outputFileConfig.getPartPrefix() + '-' + uniqueId + '-' + currentPartCounter
+						+ outputFileConfig.getPartSuffix());
+	}
+
+	private void closePartFile() throws IOException {
+		if (inProgressPart != null) {
+			InProgressFileWriter.PendingFileRecoverable pendingFileRecoverable = inProgressPart.closeForCommit();
+			pendingFiles.add(pendingFileRecoverable);
+			inProgressPart = null;
+		}
+	}
+
+	void disposePartFile() {
+		if (inProgressPart != null) {
+			inProgressPart.dispose();
+		}
+	}
+
+	// --------------------------- Testing Methods -----------------------------
+
+	@VisibleForTesting
+	public String getUniqueId() {
+		return uniqueId;
+	}
+
+	@Nullable
+	@VisibleForTesting
+	InProgressFileWriter<IN, BucketID> getInProgressPart() {
+		return inProgressPart;
+	}
+
```

Comment:
This also seems to be visible only for testing so why not adding the annotation `@VisibleForTesting`?

Reply:
Added `@VisibleForTesting` for all the testing methods.